src/reducers/query.jsx

```diff
@@ -562,6 +562,29 @@ export function removeAllFilters( state ) {
   return newState
 }
 
+/**
+* Adds a filter to the current set
+*
+* @param {object} state the current state in the Redux store
+* @param {object} action the payload containing the filter to add
+* @returns {object} the new state for the Redux store
+*/
+function addFilter( state, action ) {
```

Comment:
we should add a test for this action for the 3 different paths you can take

- has_narrative
- existing filter
- no filter existing yet

Reply:
Tests have been added.